net/net.go

```diff
@@ -1159,6 +1172,61 @@ func (n *net) createExternalLogIfNotExist(tid thread.ID, lid peer.ID, pubKey cry
 	return nil
 }
 
+// ensureUniqueLog returns a non-nil error if a log with key already exists,
+// or if a log for identity already exists for the given thread.
+func (n *net) ensureUniqueLog(id thread.ID, key crypto.Key, identity thread.PubKey) (err error) {
```

Comment:
Related to the above mentioned change, ensuring a log is unique is a bit complex. Logs can be created by specifying the the public/private key, and/or specifying the identity. This method walks through those cases.